### ISSUE TITLE
select type from @ngx-formly/material not working with grouped options

### ISSUE THREAD

**Issue by Carbammato** [formly-js/ngx-formly#953]
**I'm submitting a ...**  (check one with "x")
```
[ ] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ X ] support request
```

**Current behavior**

Hello there. Still newbie, so please don't roast me, but I've quite some trouble on generating a dropdown list where options are divided into groups.

So, I took the formly example from this link: 
https://formly-js.github.io/ngx-formly/examples/bootstrap-formly/select
Then change from bootstrap to material.

I reload the page, but it won't load, until Chrome (or Firefox) asks me to block the page.

So I'm writing to try to understand if I'm doing anything wrong.

**Expected behavior**

I simply expect a dropdown where options are divided in groups. The same I see in Material's Stackblitz example:
https://stackblitz.com/angular/nvvxerdxeqqq?file=app%2Fselect-optgroup-example.ts

**Minimal reproduction of the problem with instructions**

There you have a stackblitz example I generated:
https://stackblitz.com/edit/formly-group-select?file=src%2Fapp%2Fapp.component.ts

**What is the motivation / use case for changing the behavior?**

I gave a look to the code in github, and I noticed in ngx-formly/src/material/src/lib/types/select.ts that `get selectOptions()` is called infinite times. Let me tell you also that this not happens in case I have options without group selected.

**Please tell us about your environment:**

* **Angular version:** 6.0.X
* **Material version:** 6.0.X

Thank you in advance for your help and your work.

**Comment by thorgod** [comment]
Can you paste the code that is causing it, cant load the stackbliz because of the error.

**Comment by Carbammato** [comment]
I modified the stackblitz commenting the select form with groups and uncommenting the non-grouped one.

Anyway, I give you the code:

app.component.ts

```
import { Component } from '@angular/core';
import { FormGroup } from '@angular/forms';
import { FormlyFormOptions, FormlyFieldConfig } from '@ngx-formly/core';

@Component({
  selector: 'formly-app-example',
  templateUrl: './app.component.html',
})
export class AppComponent {
  form = new FormGroup({});
  model: any = {};
  options: FormlyFormOptions = {};

  fields: FormlyFieldConfig[] = [
    {
      key: 'marvel2',
      type: 'select',
      templateOptions: {
        label: 'Grouped Select',
        options: [
          {label: 'Iron Man', value: 'iron_man', group: 'Male'},
          {label: 'Captain America', value: 'captain_america', group: 'Male'},
          {label: 'Black Widow', value: 'black_widow', group: 'Female'},
          {label: 'Hulk', value: 'hulk', group: 'Male'},
          {label: 'Captain Marvel', value: 'captain_marvel', group: 'Female'},
        ],
      },
    },
  ];

  submit() {
    alert(JSON.stringify(this.model));
  }
}
```

app.component.html

```
<form [formGroup]="form" (ngSubmit)="submit()">
  <formly-form [model]="model" [fields]="fields" [options]="options" [form]="form">
    <button type="submit" class="btn btn-primary submit-button" [disabled]="!form.valid">Submit</button>
  </formly-form>
</form>
```

app.module.ts

```
import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { ReactiveFormsModule } from '@angular/forms';
import { FormlyModule } from '@ngx-formly/core';
import { FormlyMaterialModule } from '@ngx-formly/material';
import { MatSelectModule } from '@angular/material';

import { AppComponent } from './app.component';
@NgModule({
  imports: [
    BrowserModule,
    BrowserAnimationsModule,
    MatSelectModule,
    ReactiveFormsModule,
    FormlyMaterialModule,
    FormlyModule.forRoot(),
  ],
  bootstrap: [AppComponent],
  declarations: [
    AppComponent,
  ],
})
export class AppModule { }
```

I also added the ordinary
`@import "~@angular/material/prebuilt-themes/indigo-pink.css";`
to styles.scss

**Comment by aitboudad** [comment]
fixed in `4.1.0`
